### PR TITLE
ACM-16651: BM hosts on BM platform only

### PR DESCRIPTION
**Pull request by mareklibra**

### CODE REVIEW

File: src/cim/components/InfraEnv/InfraEnvFormPage.tsx [hunk 1]
```diff
@@ -124,6 +125,13 @@ const InfraEnvForm: React.FC<InfraEnvFormProps> = ({ onValuesChanged }) => {
                     This will determine for the infrastructure environment which kind of hosts would
                     be able to be added. If the hosts that you want to add are using DHCP server,
                     select this option, else, select the static IP.
+                    {!isBMPlatform && (
+                      // TODO(mlibra): This limitation needs to be updated once https://github.com/openshift/enhancements/pull/871 lands.
+                      <>
+                        <br />
+                        The static IPs are supported on non-bare metal platforms only.
```
Comment (rawagner) [inline src/cim/components/InfraEnv/InfraEnvFormPage.tsx:132]:
```suggestion
                        The static IPs are supported on bare metal platform only.
```

right ?

Comment (mareklibra) [reply src/cim/components/InfraEnv/InfraEnvFormPage.tsx:132]: right.

File: src/cim/components/InfraEnv/InfraEnvFormPage.tsx [hunk 2]
```diff
@@ -124,6 +125,13 @@ const InfraEnvForm: React.FC<InfraEnvFormProps> = ({ onValuesChanged }) => {
                     This will determine for the infrastructure environment which kind of hosts would
                     be able to be added. If the hosts that you want to add are using DHCP server,
                     select this option, else, select the static IP.
+                    {!isBMPlatform && (
+                      // TODO(mlibra): This limitation needs to be updated once https://github.com/openshift/enhancements/pull/871 lands.
+                      <>
+                        <br />
+                        The static IPs are supported on bare metal platform only.
```
Comment (mareklibra) [inline src/cim/components/InfraEnv/InfraEnvFormPage.tsx:132]: @matthewcarleton Can you please have a look at phrasing?

File: src/cim/components/modals/AddHostModal.tsx [hunk 3]
```diff
@@ -1,14 +1,34 @@
...
 };
 
-const AddHostModal: React.FC<AddHostModalProps> = ({ isOpen, onClose, infraEnv, onCreate }) => {
+// TODO(mlibra): This limitation needs to be updated once https://github.com/openshift/enhancements/pull/871 lands.
+const BmOnBmOnlyPopoverIcon = () => (
+  <PopoverIcon
+    noVerticalAlign
+    bodyContent={
+      <>
+        The Advanced Cluster Manager can manage bare metal hosts when deployed on bare metal
```
Comment (mareklibra) [inline src/cim/components/modals/AddHostModal.tsx:18]: @matthewcarleton Can you please have a look at phrasing?

File: src/cim/components/InfraEnv/InfraEnvFormPage.tsx (hunk 2 above)
Comment (matthewcarleton) [reply src/cim/components/InfraEnv/InfraEnvFormPage.tsx:132]: How about this?
`Static IPs are only supported on bare metal platforms.`

File: src/cim/components/modals/AddHostModal.tsx (hunk 3 above)
Comment (matthewcarleton) [reply src/cim/components/modals/AddHostModal.tsx:18]: @mareklibra can you share a screenshot of where this is in the UI?

**Pull request merged**